pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -228,7 +228,7 @@ func (c *Cluster) setLoginCredentials(password string) error {
 	}()
 
 	_, err = client.ExecuteCephCommandWithRetry(func() (string, []byte, error) {
-		output, err := client.NewCephCommand(c.context, c.clusterInfo, args).RunWithTimeout(exec.CephCommandsTimeout)
+		output, err := client.NewCephCommand(c.context, c.clusterInfo, args).RunWithTimeout(exec.CephDashboardCommandTimeout)
 		return "set dashboard creds", output, err
 	}, c.exitCode, 5, invalidArgErrorCode, dashboardInitWaitTime)
```

Comment:
2023-04-28 10:27:12.374392 E | op-mgr: failed modules: "dashboard". failed to initialize dashboard: failed to set login credentials for the ceph dashboard: failed to set login creds on mgr: failed to complete command for set dashboard creds: exec timeout waiting for the command ceph to return

Reply:
Are there five of those messages in the log? Since there are five retries currently

Reply:
```
	for i := 0; i < retries; i++ {
		action, data, err := cmd()
		if err != nil {
			exitCode, parsed := getExitCode(err)
			if parsed {
				if exitCode == retryOnExitCode {
					logger.Infof("command failed for %s. trying again...", action)
					time.Sleep(waitTime)
					continue
				}
			}
			return nil, errors.Wrapf(err, "failed to complete command for %s", action)
		}
```

Actually retries only take place  when these 2 condition match `exitCode, parsed := getExitCode(err) && if parsed {`

Otherwise, it returns,
Currently it returns from here `return nil, errors.Wrapf(err, "failed to complete command for %s", action)` So even increasing the retries also doesn't helps

Reply:
I'm afraid just increasing the timeout won't always help. We need a retry when it fails.

Reply:
If I update the full code like this then it works with the same retires and the same timeout:

```
for i := 0; i < retries; i++ {
		action, data, err := cmd()
		if err != nil {
			continue
		}
		if i > 0 {
			logger.Infof("action %s succeeded on attempt %d", action, i)
		}
		return data, nil
	}
	return nil, errors.New("max command retries exceeded")
}
```

Reply:
PLease review latest code

Reply:
Now it retries